spec/src/main/asciidoc/metrics.asciidoc

```diff
@@ -26,164 +26,188 @@ the methods annotated with a `@Retry`, `@Timeout`, `@CircuitBreaker`, `@Bulkhead
-| Counter | None
-| The number of times the method timed out
-
-|`ft.<name>.timeout.callsNotTimedOut.total`
-| Counter | None
-| The number of times the method completed without timing out
+| Name | `ft.timeout.calls.total`
+
+| Type | `Counter`
+| Unit | None
+| Description | The number of times the timeout logic was run. This will usually be once per method call, but may be zero times if the circuit breaker prevents execution or more than once if the method is retried.
+| Tags
+a| * `method` - the fully qualified method name
+* `timedOut` = `[true\|false]` - whether the method call timed out
 |===
 
-Note that the sum of `ft.<name>.timeout.callsTimedOut.total` and `ft.<name>.timeout.callsNotTimedOut.total` will give the total number of calls for which the Timeout logic was run. This may be larger than the total number of invocations of the method it's also annotated with `@Retry` because the Timeout logic is applied to each Retry attempt, not to the whole method invocation time.
-
-=== Metrics added for `@CircuitBreaker`
-
-[cols="8,3,4,9"]
+[cols="1,5"]
 |===
-| Name | Type | Unit | Description
+| Name | `ft.timeout.executionDuration`
 
-|`ft.<name>.circuitbreaker.callsSucceeded.total`
-| Counter | None
-| Number of calls which were allowed to run and were considered a success by the circuit breaker
-
-|`ft.<name>.circuitbreaker.callsFailed.total`
-| Counter | None
-| Number of calls which were allowed to run and were considered a failure by the circuit breaker
-
-|`ft.<name>.circuitbreaker.callsPrevented.total`
-| Counter | None
-| Number of calls which were prevented from running by the circuit breaker
-
-|`ft.<name>.circuitbreaker.open.total`
-| Gauge<Long> | Nanoseconds
-| Amount of time the circuit breaker has spent in open state
+| Type | `Histogram`
+| Unit | Nanoseconds
+| Description | Histogram of execution times for the method
+| Tags
+a| * `method` - the fully qualified method name
+|===
 
-|`ft.<name>.circuitbreaker.halfOpen.total`
-| Gauge<Long> | Nanoseconds
-| Amount of time the circuit breaker has spent in half-open state
+=== Metrics added for `@CircuitBreaker`
 
-|`ft.<name>.circuitbreaker.closed.total`
-| Gauge<Long> | Nanoseconds
-| Amount of time the circuit breaker has spent in closed state
+ft.circuitbreaker.calls.total{method=<name>, circuitBreakerResult=[success|failure|circuitBreakerOpen]}
 
-|`ft.<name>.circuitbreaker.opened.total`
-| Counter | None
-| Number of times the circuit breaker has moved from closed state to open state
+[cols="1,5"]
+|===
+| Name | `ft.circuitbreaker.calls.total`
+
+| Type | `Counter`
+| Unit | None
+| Description | The number of times the timeout logic was run. This will usually be once per method call, but may be more than once if the method call is retried.
+| Tags
+a| * `method` - the fully qualified method name
+ * `circuitBreakerResult` = `[success\|failure\|circuitBreakerOpen]` - the result of the method call, as considered by the circuit breaker logic
```

Comment:
Need to add more explaintation on the success criteria.

In more details, the method call is considered to be successful when an method throws an exception but the exception is excluded from the `CircuitBreaker` based on its skipOn and 'failOn` criteria.

Reply:
IMHO, we don't need to repeat that here. That's clearly defined in the circuit breaker chapter.

Reply:
Maybe we could just reference back to the section where it is defined?

Reply:
I've added a cross reference.